### PR TITLE
feat: Add analytics metadata to Side navigation

### PR DESCRIPTION
**Pull request by fralongo**
### Description

Similar to https://github.com/cloudscape-design/components/commit/2e32ba37c3fb6c6e8735dfe88d713735c7d61cb9

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/side-navigation/parts.tsx [hunk 1]
```diff
@@ -79,7 +97,14 @@ interface Item {
...
 }
 
-export function NavigationItemsList({ items, variant, activeHref, fireChange, fireFollow }: NavigationItemsListProps) {
+export function NavigationItemsList({
+  items,
+  variant,
+  activeHref,
+  fireChange,
+  fireFollow,
+  position = '',
```
Comment (just-boris) [inline src/side-navigation/parts.tsx:106]: When is this one undefined?

Comment (fralongo) [reply src/side-navigation/parts.tsx:106]: Here: https://github.com/cloudscape-design/components/blob/main/src/side-navigation/implementation.tsx#L75

Comment (just-boris) [reply src/side-navigation/parts.tsx:106]: Would it be better to have an explicit "root" position?

Comment (fralongo) [reply src/side-navigation/parts.tsx:106]: `position` is chained: https://github.com/cloudscape-design/components/pull/2725/files#diff-08a958e7ca3d2a2f7a1bb81060fe998806ac3848e8f12c59a121a8707a47dfcaR117

**Review by just-boris — APPROVED** [review]

File: src/side-navigation/parts.tsx (hunk 1 above)
Comment (just-boris) [reply src/side-navigation/parts.tsx:106]: Still find it confusing, but okay

**Pull request merged**